### PR TITLE
Added implementation of the Max flow algorithm (Dinic) and Topological sorting

### PR DESCRIPTION
**Pull request by PalAditya**
Thanks for merging my first PR. I would like to keep contributing, if that's okay? :)
Please tel me about any mistakes though.

### CODE REVIEW

**Comment by christianbender** [comment]
@PalAditya  
> Thanks for merging my first PR. I would like to keep contributing, if that's okay? :)
of course :+1:

File: DataStructures/Graphs/MaxFlowDinic.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
+import java.util.*;
+class MaxFlowDinic
+{
+    private int V,level[];
+    private ArrayList<Edge> adj[];
+    class Edge
```
Comment (christianbender) [inline DataStructures/Graphs/MaxFlowDinic.java]: In latest version of Java (java 8) you don't need inner classes. You can write this sperated in one single file like:  
```java
// one file
class A {

}

class B {

}

// ... 
```

File: DataStructures/Graphs/MaxFlowDinic.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        g.addEdge(1, 2, 10 );
+        g.addEdge(1, 3, 12 );
+        g.addEdge(2, 1, 4 );
+        g.addEdge(2, 4, 14);
+        g.addEdge(3, 2, 9 );
+        g.addEdge(3, 5, 20 );
+        g.addEdge(4, 3, 7 );
+        g.addEdge(4, 5, 4);
+        System.out.println( "Maximum flow is: "+g.DinicMaxflow(0, 5));    
+    }
```
Comment (christianbender) [inline DataStructures/Graphs/MaxFlowDinic.java]: If I compile this I get:
> Note: MaxFlowDinic.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

You must use type specifies ```<TYPE>``` by each collection.

File: DataStructures/Graphs/MaxFlowDinic.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        Edge a=new Edge(v, 0, C, adj[v].size());// Forward edge : 0 flow and C capacity       
+        Edge b=new Edge(u, 0, 0, adj[u].size());// Back edge : 0 flow and 0 capacity
+        adj[u].add(a);
+        adj[v].add(b); // reverse edge
+    }
+    /**
+     * @param s=Source vertex
+     * @param t=Sink Vertex
+     */
+    private boolean BFS(int s,int t)
```
Comment (christianbender) [inline DataStructures/Graphs/MaxFlowDinic.java]: Please use java naming conventions (camelCase)

File: DataStructures/Graphs/MaxFlowDinic.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        level=new int[V];
+    }
+    /**
+     * @param u Start vertex
+     * @param v end vertex
+     * @param C Capacity of edge
+    */
+    void addEdge(int u, int v, double C)
+    {
+        Edge a=new Edge(v, 0, C, adj[v].size());// Forward edge : 0 flow and C capacity
```
Comment (christianbender) [inline DataStructures/Graphs/MaxFlowDinic.java]: Please put in spaces between the operators like ```=```

File: DataStructures/Graphs/MaxFlowDinic.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+                    break;
+                total += flow;
+            }
+        }
+        return total;
+    }
+    public static void main(String args[])
+    {
+        MaxFlowDinic obj=new MaxFlowDinic(0);//Dummy object to access the non-static methods
+        obj.go();
```
Comment (christianbender) [inline DataStructures/Graphs/MaxFlowDinic.java]: Put in the expecting output.

File: DataStructures/Graphs/MaxFlowDinic.java [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+        return 0;
+    }
+    /**
+     * @param s=Source vertex
+     * @param t=Sink Vertex
+     */
+    double DinicMaxflow(int s, int t)
+    {
+        // Corner case
+        if (s == t)
```
Comment (christianbender) [inline DataStructures/Graphs/MaxFlowDinic.java]: Use curly braces.

File: DataStructures/Graphs/TopologicalSort.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    public static void main(String args[])
+    {
+        TopologicalSort g = new TopologicalSort(6);
+        g.addEdge(5, 2);
+        g.addEdge(5, 0);
+        g.addEdge(4, 0);
+        g.addEdge(4, 1);
+        g.addEdge(2, 3);
+        g.addEdge(3, 1);
+        g.topologicalSort();
```
Comment (christianbender) [inline DataStructures/Graphs/TopologicalSort.java]: Add comments for the expected output.

File: DataStructures/Graphs/TopologicalSort.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    }
+}
```
Comment (christianbender) [inline DataStructures/Graphs/TopologicalSort.java]: Same warning as above.
> Note: TopologicalSort.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

File: DataStructures/Graphs/TopologicalSort.java [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+     */
+    public void addEdge(int v,int w)
+    { 
+        adj[v].add(w);//Unidirectional, as topological sort is meaningless for bidirectional/undirected graphs
+    }
+    // A recursive function used by topologicalSort
+    private void topologicalSortUtil(int v, boolean visited[], Stack stack)
+    {       
+        visited[v] = true;// Marking the current node as visited.
+        Integer i;
```
Comment (christianbender) [inline DataStructures/Graphs/TopologicalSort.java]: Write instead ```int i = 0;```

File: DataStructures/Graphs/TopologicalSort.java [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+    {         
+        // Mark the current node as visited and part of recursion stack
+        if (recStack[i])
+            return true;//Condition for cycle
+        if (visited[i])
+            return false; //No need to continue DFSing on visoted node so return          
+        visited[i] = true;
+        recStack[i] = true;
+        List<Integer> children = adj[i];         
+        for (Integer c: children)
```
Comment (christianbender) [inline DataStructures/Graphs/TopologicalSort.java]: Write ```int c```

**Comment by varunu28** [comment]
@PalAditya  Can you please do the changes requested? Otherwise you can close this PR and open a fresh one with changes implemented later

**Comment by PalAditya** [comment]
Made the requested changes

**Comment by barchiye** [comment]
hello everyone ....hope all of u are fine ...i just want to get the source code of modified throttled algorithm for load balancing algorithm in cloud computing

**Comment by ayaankhan98** [comment]
@PalAditya any update over this?

**Pull request closed without merging**